common/lib/xmodule/xmodule/modulestore/split_mongo/mongo_connection.py

```diff
@@ -203,6 +209,40 @@ def structure_to_mongo(structure, course_context=None):
         return new_structure
 
 
+class CourseStructureCache(object):
+    """
+    Wrapper around django cache object to cache course structure objects.
+    The course structures are pickled and compressed when cached.
+    """
+    def __init__(self):
+        try:
+            self.cache = get_cache('course_structure_cache')
+        except InvalidCacheBackendError:
+            self.cache = get_cache('default')
```

Comment:
I wasn't sure about using this or just not caching. This approach seemed like it could lead to unexpectedly bad cache behavior (if a site isn't ready for it), but I guess the other approach could lead to unexpectedly bad mongo behavior, so maybe it's a wash.